ostree/ostree_transport.go

```diff
@@ -87,21 +87,13 @@ func NewReference(image string, repo string) (types.ImageReference, error) {
 	// We use the reference.* parsers basically for the default tag name in
 	// reference.TagNameOnly, and incidentally for some character set and length
 	// restrictions.
-	var ostreeImage reference.Named
-	s := strings.SplitN(image, ":", 2)
-
-	named, err := reference.WithName(s[0])
+	ostreeImage, err := reference.ParseNormalizedNamed(image)
```

Comment:
Fair enough; please update the comment to say ”to detect whether the reference has a tag” (the code below hard-codes the default value now).

Non-blocking: Consider also emphasizing that the result of the parsing is ignored (in the comment above or below when creating the `ostreeRefrence`), so the normalization done does not matter, and naming the variable something which sounds less relevant (`ostreeImage` seems more pertinent than the input `image`, but the code is intentionally throwing it away), maybe `parsedRef` or something like that.